### ISSUE TITLE
Error inflating class com.rey.material.widget.Spinner .

### ISSUE THREAD

**Issue by HimanshuLakra** [rey5137/material#280]
Its working fine but when I made changes in gradle file like change version of appcompat to 23.1.0 etc the application just crashed.

Logcat

10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime: android.view.InflateException: Binary XML file line #181: Error inflating class com.rey.material.widget.Spinner
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createView(LayoutInflater.java:613)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rocket.freeloader.Draw.onCreateView(Draw.java:70)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:1965)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1078)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1259)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1624)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:517)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:615)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:92)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:155)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5520)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invokeNative(Native Method)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Method.java:511)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1029)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:796)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at dalvik.system.NativeStart.main(Native Method)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:  Caused by: java.lang.reflect.InvocationTargetException
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Constructor.constructNative(Native Method)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createView(LayoutInflater.java:587)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rocket.freeloader.Draw.onCreateView(Draw.java:70) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:1965) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1078) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1259) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1624) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:517) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:615) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:92) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:155) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5520) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invokeNative(Native Method) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Method.java:511) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1029) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:796) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at dalvik.system.NativeStart.main(Native Method) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:  Caused by: java.lang.NoSuchFieldError: com.rey.material.R$styleable.Spinner_android_gravity
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rey.material.widget.Spinner.applyStyle(Spinner.java:234)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rey.material.widget.Spinner.init(Spinner.java:149)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rey.material.widget.Spinner.<init>(Spinner.java:128)
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Constructor.constructNative(Native Method) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Constructor.newInstance(Constructor.java:417) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createView(LayoutInflater.java:587) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.rocket.freeloader.Draw.onCreateView(Draw.java:70) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:1965) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1078) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1259) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1624) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:517) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:615) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:92) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:155) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5520) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invokeNative(Native Method) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Method.java:511) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1029) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:796) 
10-19 23:22:53.381 1330-1330/com.rocket.freeloader E/AndroidRuntime:     at dalvik.system.NativeStart.main(Native Method) 

Gradle: 
![gradle1](https://cloud.githubusercontent.com/assets/7153301/10635078/bdd6c77c-7813-11e5-9581-01794a87744c.png)
![gradle2](https://cloud.githubusercontent.com/assets/7153301/10635086/d1159cc8-7813-11e5-94ca-3f6ad7d0b0a6.png)

**Comment by PsyGik** [comment]
This was reported earlier. Please use the snapshot version.

Update `compile 'com.github.rey5137:material:1.2.1'`
to `compile 'com.github.rey5137:material:1.2.1.6-SNAPSHOT'`

Add the snapshot repo also,

    repositories { maven { url "https://oss.sonatype.org/content/repositories/snapshots" } }

**Comment by HimanshuLakra** [comment]
After using snapshot , it is showing this error.
Error:Execution failed for task ':app:transformClassesWithJarMergingForDebug'.
> com.android.build.transform.api.TransformException: java.util.zip.ZipException: duplicate entry: com/rey/material/app/BottomSheetDialog$1.class

**Comment by PsyGik** [comment]
Duplicate entry. Make sure you're including ONLY the snapshot version and not both. Clean the project and try building.

**Comment by HimanshuLakra** [comment]
![gradle1](https://cloud.githubusercontent.com/assets/7153301/10693158/34b64074-79b4-11e5-9268-ebac859fdb22.png)
I have included it only once.

**Comment by NagoorBabu** [comment]
Even after using snapshot version, Application crashing. How to fix this issue?

**Comment by townindore** [comment]
Having the same issue, this library doesn't seem to work with 23.0.1 at all, any solutions?

**Comment by 4nd3250n** [comment]
Yes, still not resolved.

**Comment by wandering7man** [comment]
Having the same issue

**Comment by 4nd3250n** [comment]
@wandering7man 
Try to use aforementioned snapshot 1.2.1.6. I have managed to compile library using 1.2.1.9.
But since I am using eclipse I could not just add dependency. I had to go to snapshot repository:
https://oss.sonatype.org/content/repositories/snapshots/com/github/rey5137/material/
Then I looked for 1.2.1.9
https://oss.sonatype.org/content/repositories/snapshots/com/github/rey5137/material/1.2.1.9-SNAPSHOT/
And then I took the content of this two files:
material-1.2.1.9-20150916.022521-1-sources.jar
material-1.2.1.9-20150916.022521-1.aar

Unzipped them and combined what I found in there into an eclipse project. And at the moment I am at testing it.
